src/bsdfs/roughconductor.cpp

```diff
@@ -274,6 +277,10 @@ class RoughConductor final : public BSDF<Float, Spectrum> {
             F = fresnel_conductor(UnpolarizedSpectrum(dot(si.wi, m)), eta_c);
         }
 
+        /* If requested, include the specular reflectance component */
+        if (m_specular_reflectance)
+            F *= unpolarized<Spectrum>(m_specular_reflectance->eval(si, active));
```

Comment:
Change `weight` above to type `UnpolarizedSpectrum` & change this line to ` weight *= m_specular_reflectance->eval(si, active);
`.